lib/DeonRestClient.ts

```diff
@@ -96,10 +98,13 @@ const possiblyBadRequestOrNotFound = async (r: Response) => {
   throwUnexpected(r, data);
 };
 
-const checkHandler = async (r: Response): Promise<CheckError[]> => {
-  if (r.ok) { return []; }
+const checkHandler = async (r: Response): Promise<CheckResponse> => {
+  if (r.ok) {
+    return r.json().then((warnings: Warning[]) =>
+      ({ warnings, errors: [] }));
+  }
   if (r.status === 400) {
-    return r.json().then((json: {errors: CheckError[]}) => json.errors);
+    return r.json().then((rsp: { warnings: Warning[], errors: CheckError[] }) => rsp);
```

Comment:
Can't you cast to a `CheckResponse` here instead?
```suggestion
    return r.json().then((rsp: CheckResponse) => rsp);
```